build failing due to missing cmsis_core.h

Originally I created a new issue here: https://github.com/embeddedcontainers/examples-zephyr-helloworld/issues/3
But then I also checked with ghcr.io/embeddedcontainers/zephyr:arm-0.17.0SDK directly and it has the same problem.

Since 4-May-2025 the build (from scratch) is not working anymore due to probably not including a directory anymore:

    root@a4b942fd9bd9:/workspace# west build -b nucleo_f429zi app -c
    -- west build: generating a build system
    Loading Zephyr default modules (Zephyr base (cached)).
    -- Application: /workspace/app
    -- CMake version: 3.25.1
    -- Cache files will be written to: /root/.cache/zephyr
    -- Zephyr version: 4.1.99 (/workspace/deps/zephyr)
    -- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
    -- Board: nucleo_f429zi, qualifiers: stm32f429xx
    -- Found host-tools: zephyr 0.16.4 (/opt/toolchains/zephyr-sdk-0.16.4)
    -- Found toolchain: zephyr 0.16.4 (/opt/toolchains/zephyr-sdk-0.16.4)
    -- Found BOARD.dts: /workspace/deps/zephyr/boards/st/nucleo_f429zi/nucleo_f429zi.dts
    -- Generated zephyr.dts: /workspace/build/zephyr/zephyr.dts
    -- Generated pickled edt: /workspace/build/zephyr/edt.pickle
    -- Generated devicetree_generated.h: /workspace/build/zephyr/include/generated/zephyr/devicetree_generated.h
    Parsing /workspace/deps/zephyr/Kconfig
    Loaded configuration '/workspace/build/zephyr/.config'
    No change to configuration in '/workspace/build/zephyr/.config'
    No change to Kconfig header in '/workspace/build/zephyr/include/generated/zephyr/autoconf.h'
    -- Found gen_kobject_list: /workspace/deps/zephyr/scripts/build/gen_kobject_list.py
    -- Configuring done
    -- Generating done
    -- Build files have been written to: /workspace/build
    -- west build: building application
    [2/132] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
    FAILED: zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
    /opt/toolchains/zephyr-sdk-0.16.4/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DCORE_CM4 -DHSE_VALUE=8000000 -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DPICOLIBC_LONG_LONG_PRINTF_SCANF -DSTM32F429xx -DUSE_FULL_LL_DRIVER -DUSE_HAL_DRIVER -D__LINUX_ERRNO_EXTENSIONS__ -D__ZEPHYR__=1 -I/workspace/deps/zephyr/kernel/include -I/workspace/deps/zephyr/arch/arm/include -I/workspace/build/zephyr/include/generated/zephyr -I/workspace/deps/zephyr/include -I/workspace/build/zephyr/include/generated -I/workspace/deps/zephyr/soc/st/stm32 -I/workspace/deps/zephyr/soc/st/stm32/common/. -I/workspace/deps/zephyr/drivers -I/workspace/deps/zephyr/soc/st/stm32/stm32f4x/. -I/workspace/deps/modules/hal/stm32/stm32cube/stm32f4xx/soc -I/workspace/deps/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include -I/workspace/deps/modules/hal/stm32/stm32cube/common_ll/include -isystem /workspace/deps/zephyr/lib/libc/common/include -fno-strict-aliasing -Os -imacros /workspace/build/zephyr/include/generated/zephyr/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m4 -mthumb -mabi=aapcs -mfp16-format=ieee -mtp=soft --sysroot=/opt/toolchains/zephyr-sdk-0.16.4/arm-zephyr-eabi/arm-zephyr-eabi -imacros /workspace/deps/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/workspace/app=CMAKE_SOURCE_DIR -fmacro-prefix-map=/workspace/deps/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/workspace=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=picolibc.specs -std=c99 -fno-lto -MD -MT zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj -MF zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj.d -o zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj -c /workspace/deps/zephyr/arch/arm/core/offsets/offsets.c
    In file included from /workspace/deps/zephyr/include/zephyr/arch/arm/asm_inline.h:18,
    from /workspace/deps/zephyr/include/zephyr/arch/arm/arch.h:30,
    from /workspace/deps/zephyr/include/zephyr/arch/cpu.h:19,
    from /workspace/deps/zephyr/include/zephyr/kernel_includes.h:36,
    from /workspace/deps/zephyr/include/zephyr/kernel.h:17,
    from /workspace/deps/zephyr/arch/arm/core/offsets/offsets_aarch32.c:28,
    from /workspace/deps/zephyr/arch/arm/core/offsets/offsets.c:9:
    /workspace/deps/zephyr/include/zephyr/arch/arm/asm_inline_gcc.h:24:10: fatal error: cmsis_core.h: No such file or directory
    24 | #include <cmsis_core.h>
    | ^~~~~~~~~~~~~~
    compilation terminated.
    ninja: build stopped: subcommand failed.
    FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /workspace/build

It can be reproduced by

    cloning: https://github.com/embeddedcontainers/examples-zephyr-helloworld
    start vscode in this folder
    ctrl+shift+p; reopen on container
    execute in terminal: west build -b nucleo_f429zi app -c

The missing file does exist in the next directories:
./deps/zephyr/modules/cmsis_6/cmsis_core.h
./deps/zephyr/modules/cmsis/cmsis_core.h

When adding modules/cmsis in the file /workspace/deps/zephyr/CMakeLists.txt (not sure if this is even the correct location to be added)

    target_include_directories(${OFFSETS_LIB} PRIVATE
    kernel/include
    modules/cmsis
    ${ARCH_DIR}/${ARCH}/include
    )

Then the compilation continues with many warnings and errors:
/workspace/deps/zephyr/arch/arm/include/cortex_m/kernel_arch_func.h:89:9: error: 'SCB' undeclared (first use in this function)
89 | SCB->ICSR |= SCB_ICSR_PENDSVSET_Msk;

    root@b1b842669914:/workspace# west build -b nucleo_f429zi app
    [0/1] Re-running CMake...
    Loading Zephyr default modules (Zephyr base (cached)).
    -- Application: /workspace/app
    -- CMake version: 3.25.1
    -- Cache files will be written to: /root/.cache/zephyr
    -- Zephyr version: 4.1.99 (/workspace/deps/zephyr)
    -- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
    -- Board: nucleo_f429zi, qualifiers: stm32f429xx
    -- Found host-tools: zephyr 0.16.4 (/opt/toolchains/zephyr-sdk-0.16.4)
    -- Found toolchain: zephyr 0.16.4 (/opt/toolchains/zephyr-sdk-0.16.4)
    -- Found BOARD.dts: /workspace/deps/zephyr/boards/st/nucleo_f429zi/nucleo_f429zi.dts
    -- Generated zephyr.dts: /workspace/build/zephyr/zephyr.dts
    -- Generated pickled edt: /workspace/build/zephyr/edt.pickle
    -- Generated devicetree_generated.h: /workspace/build/zephyr/include/generated/zephyr/devicetree_generated.h
    Parsing /workspace/deps/zephyr/Kconfig
    Loaded configuration '/workspace/build/zephyr/.config'
    No change to configuration in '/workspace/build/zephyr/.config'
    No change to Kconfig header in '/workspace/build/zephyr/include/generated/zephyr/autoconf.h'
    -- Found gen_kobject_list: /workspace/deps/zephyr/scripts/build/gen_kobject_list.py
    -- Configuring done
    -- Generating done
    -- Build files have been written to: /workspace/build
    [2/132] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
    FAILED: zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
    /opt/toolchains/zephyr-sdk-0.16.4/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DCORE_CM4 -DHSE_VALUE=8000000 -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DPICOLIBC_LONG_LONG_PRINTF_SCANF -DSTM32F429xx -DUSE_FULL_LL_DRIVER -DUSE_HAL_DRIVER -D__LINUX_ERRNO_EXTENSIONS__ -D__ZEPHYR__=1 -I/workspace/deps/zephyr/kernel/include -I/workspace/deps/zephyr/modules/cmsis -I/workspace/deps/zephyr/arch/arm/include -I/workspace/build/zephyr/include/generated/zephyr -I/workspace/deps/zephyr/include -I/workspace/build/zephyr/include/generated -I/workspace/deps/zephyr/soc/st/stm32 -I/workspace/deps/zephyr/soc/st/stm32/common/. -I/workspace/deps/zephyr/drivers -I/workspace/deps/zephyr/soc/st/stm32/stm32f4x/. -I/workspace/deps/modules/hal/stm32/stm32cube/stm32f4xx/soc -I/workspace/deps/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include -I/workspace/deps/modules/hal/stm32/stm32cube/common_ll/include -isystem /workspace/deps/zephyr/lib/libc/common/include -fno-strict-aliasing -Os -imacros /workspace/build/zephyr/include/generated/zephyr/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m4 -mthumb -mabi=aapcs -mfp16-format=ieee -mtp=soft --sysroot=/opt/toolchains/zephyr-sdk-0.16.4/arm-zephyr-eabi/arm-zephyr-eabi -imacros /workspace/deps/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/workspace/app=CMAKE_SOURCE_DIR -fmacro-prefix-map=/workspace/deps/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/workspace=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=picolibc.specs -std=c99 -fno-lto -MD -MT zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj -MF zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj.d -o zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj -c /workspace/deps/zephyr/arch/arm/core/offsets/offsets.c
    In file included from /workspace/deps/zephyr/include/zephyr/arch/arm/asm_inline.h:18,
    from /workspace/deps/zephyr/include/zephyr/arch/arm/arch.h:30,
    from /workspace/deps/zephyr/include/zephyr/arch/cpu.h:19,
    from /workspace/deps/zephyr/include/zephyr/kernel_includes.h:36,
    from /workspace/deps/zephyr/include/zephyr/kernel.h:17,
    from /workspace/deps/zephyr/arch/arm/core/offsets/offsets_aarch32.c:28,
    from /workspace/deps/zephyr/arch/arm/core/offsets/offsets.c:9:
    /workspace/deps/zephyr/include/zephyr/arch/arm/asm_inline_gcc.h: In function 'arch_irq_lock':
    /workspace/deps/zephyr/include/zephyr/arch/arm/asm_inline_gcc.h:56:15: warning: implicit declaration of function '__get_BASEPRI' [-Wimplicit-function-declaration]
    56 | key = __get_BASEPRI();
    | ^~~~~~~~~~~~~
    /workspace/deps/zephyr/include/zephyr/arch/arm/asm_inline_gcc.h:57:9: warning: implicit declaration of function '__set_BASEPRI_MAX' [-Wimplicit-function-declaration]
    57 | __set_BASEPRI_MAX(_EXC_IRQ_DEFAULT_PRIO);
    | ^~~~~~~~~~~~~~~~~
    /workspace/deps/zephyr/include/zephyr/arch/arm/asm_inline_gcc.h:58:9: warning: implicit declaration of function '__ISB' [-Wimplicit-function-declaration]
    58 | __ISB();
    | ^~~~~
    /workspace/deps/zephyr/include/zephyr/arch/arm/asm_inline_gcc.h: In function 'arch_irq_unlock':
    /workspace/deps/zephyr/include/zephyr/arch/arm/asm_inline_gcc.h:89:9: warning: implicit declaration of function '__set_BASEPRI' [-Wimplicit-function-declaration]
    89 | __set_BASEPRI(key);
    | ^~~~~~~~~~~~~
    In file included from /workspace/deps/zephyr/arch/arm/include/kernel_arch_data.h:28,
    from /workspace/deps/zephyr/arch/arm/core/offsets/offsets_aarch32.c:29:
    /workspace/deps/zephyr/arch/arm/include/cortex_m/stack.h: In function 'z_arm_interrupt_stack_setup':
    /workspace/deps/zephyr/arch/arm/include/cortex_m/stack.h:44:9: warning: implicit declaration of function '__set_MSP' [-Wimplicit-function-declaration]
    44 | __set_MSP(msp);
    | ^~~~~~~~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/stack.h:60:9: error: 'SCB' undeclared (first use in this function)
    60 | SCB->CCR |= SCB_CCR_STKALIGN_Msk;
    | ^~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/stack.h:60:9: note: each undeclared identifier is reported only once for each function it appears in
    /workspace/deps/zephyr/arch/arm/include/cortex_m/stack.h:60:21: error: 'SCB_CCR_STKALIGN_Msk' undeclared (first use in this function)
    60 | SCB->CCR |= SCB_CCR_STKALIGN_Msk;
    | ^~~~~~~~~~~~~~~~~~~~
    In file included from /workspace/deps/zephyr/arch/arm/include/kernel_arch_data.h:29:
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h: In function 'arch_is_in_isr':
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:95:17: warning: implicit declaration of function '__get_IPSR' [-Wimplicit-function-declaration]
    95 | return (__get_IPSR()) ? (true) : (false);
    | ^~~~~~~~~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h: In function 'arch_is_in_nested_exception':
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:120:35: error: 'IPSR_ISR_Msk' undeclared (first use in this function)
    120 | return (esf->basic.xpsr & IPSR_ISR_Msk) ? (true) : (false);
    | ^~~~~~~~~~~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h: In function 'z_arm_exc_setup':
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:149:9: warning: implicit declaration of function 'NVIC_SetPriority' [-Wimplicit-function-declaration]
    149 | NVIC_SetPriority(PendSV_IRQn, _EXC_PENDSV_PRIO);
    | ^~~~~~~~~~~~~~~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:149:26: error: 'PendSV_IRQn' undeclared (first use in this function)
    149 | NVIC_SetPriority(PendSV_IRQn, _EXC_PENDSV_PRIO);
    | ^~~~~~~~~~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:155:26: error: 'SVCall_IRQn' undeclared (first use in this function)
    155 | NVIC_SetPriority(SVCall_IRQn, _EXC_SVC_PRIO);
    | ^~~~~~~~~~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:159:26: error: 'MemoryManagement_IRQn' undeclared (first use in this function)
    159 | NVIC_SetPriority(MemoryManagement_IRQn, _EXC_FAULT_PRIO);
    | ^~~~~~~~~~~~~~~~~~~~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:160:26: error: 'BusFault_IRQn' undeclared (first use in this function)
    160 | NVIC_SetPriority(BusFault_IRQn, _EXC_FAULT_PRIO);
    | ^~~~~~~~~~~~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:161:26: error: 'UsageFault_IRQn' undeclared (first use in this function)
    161 | NVIC_SetPriority(UsageFault_IRQn, _EXC_FAULT_PRIO);
    | ^~~~~~~~~~~~~~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:165:26: error: 'DebugMonitor_IRQn' undeclared (first use in this function)
    165 | NVIC_SetPriority(DebugMonitor_IRQn, _EXC_FAULT_PRIO);
    | ^~~~~~~~~~~~~~~~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:172:9: error: 'SCB' undeclared (first use in this function)
    172 | SCB->SHCSR |=
    | ^~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:173:17: error: 'SCB_SHCSR_USGFAULTENA_Msk' undeclared (first use in this function)
    173 | SCB_SHCSR_USGFAULTENA_Msk | SCB_SHCSR_MEMFAULTENA_Msk | SCB_SHCSR_BUSFAULTENA_Msk;
    | ^~~~~~~~~~~~~~~~~~~~~~~~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:173:45: error: 'SCB_SHCSR_MEMFAULTENA_Msk' undeclared (first use in this function)
    173 | SCB_SHCSR_USGFAULTENA_Msk | SCB_SHCSR_MEMFAULTENA_Msk | SCB_SHCSR_BUSFAULTENA_Msk;
    | ^~~~~~~~~~~~~~~~~~~~~~~~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:173:73: error: 'SCB_SHCSR_BUSFAULTENA_Msk' undeclared (first use in this function)
    173 | SCB_SHCSR_USGFAULTENA_Msk | SCB_SHCSR_MEMFAULTENA_Msk | SCB_SHCSR_BUSFAULTENA_Msk;
    | ^~~~~~~~~~~~~~~~~~~~~~~~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h: In function 'z_arm_clear_faults':
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:218:9: error: 'SCB' undeclared (first use in this function)
    218 | SCB->CFSR = SCB_CFSR_USGFAULTSR_Msk | SCB_CFSR_MEMFAULTSR_Msk | SCB_CFSR_BUSFAULTSR_Msk;
    | ^~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:218:21: error: 'SCB_CFSR_USGFAULTSR_Msk' undeclared (first use in this function)
    218 | SCB->CFSR = SCB_CFSR_USGFAULTSR_Msk | SCB_CFSR_MEMFAULTSR_Msk | SCB_CFSR_BUSFAULTSR_Msk;
    | ^~~~~~~~~~~~~~~~~~~~~~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:218:47: error: 'SCB_CFSR_MEMFAULTSR_Msk' undeclared (first use in this function)
    218 | SCB->CFSR = SCB_CFSR_USGFAULTSR_Msk | SCB_CFSR_MEMFAULTSR_Msk | SCB_CFSR_BUSFAULTSR_Msk;
    | ^~~~~~~~~~~~~~~~~~~~~~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/exception.h:218:73: error: 'SCB_CFSR_BUSFAULTSR_Msk' undeclared (first use in this function)
    218 | SCB->CFSR = SCB_CFSR_USGFAULTSR_Msk | SCB_CFSR_MEMFAULTSR_Msk | SCB_CFSR_BUSFAULTSR_Msk;
    | ^~~~~~~~~~~~~~~~~~~~~~~
    In file included from /workspace/deps/zephyr/arch/arm/include/kernel_arch_func.h:26,
    from /workspace/deps/zephyr/kernel/include/kernel_arch_interface.h:682,
    from /workspace/deps/zephyr/kernel/include/kernel_internal.h:18,
    from /workspace/deps/zephyr/kernel/include/kernel_offsets.h:8,
    from /workspace/deps/zephyr/arch/arm/core/offsets/offsets_aarch32.c:30:
    /workspace/deps/zephyr/arch/arm/include/cortex_m/kernel_arch_func.h: In function 'arch_swap':
    /workspace/deps/zephyr/arch/arm/include/cortex_m/kernel_arch_func.h:89:9: error: 'SCB' undeclared (first use in this function)
    89 | SCB->ICSR |= SCB_ICSR_PENDSVSET_Msk;
    | ^~~
    /workspace/deps/zephyr/arch/arm/include/cortex_m/kernel_arch_func.h:89:22: error: 'SCB_ICSR_PENDSVSET_Msk' undeclared (first use in this function)
    89 | SCB->ICSR |= SCB_ICSR_PENDSVSET_Msk;
    | ^~~~~~~~~~~~~~~~~~~~~~
    ninja: build stopped: subcommand failed.
    FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /workspace/build

This makes me think it is not just some include problem on the level I was trying to fix, but something an a bigger scale.

Please remember that this started on 4-May-2025. I however don't know what package was updated then to break the build.

--- a/.devcontainer/devcontainer.json
+++ b/.devcontainer/devcontainer.json
@@ -1,5 +1,5 @@
 {
-  "image": "ghcr.io/embeddedcontainers/zephyr:arm-0.16.4SDK",
+  "image": "ghcr.io/embeddedcontainers/zephyr:arm-0.17.1SDK",
   "workspaceMount": "source=${localWorkspaceFolder},target=/workspace/app,type=bind",
   "workspaceFolder": "/workspace",
   "onCreateCommand": "west init -l app/ && west update --narrow -o=--depth=1 && west zephyr-export && pip install -r deps/zephyr/scripts/requirements-base.txt",
@@ -12,7 +12,7 @@
         // Hush CMake
         "cmake.configureOnOpen": false,
         // IntelliSense
-        "C_Cpp.default.compilerPath": "/opt/toolchains/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc",
+        "C_Cpp.default.compilerPath": "/opt/toolchains/zephyr-sdk-0.17.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc",
         "C_Cpp.default.compileCommands": "build/compile_commands.json",
         // File Associations
         "files.associations": {}

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -4,10 +4,10 @@ on:
   workflow_dispatch:
     inputs:
       ZEPHYR_SDK:
-        description: "Zephyr SDK version. Ex. 0.16.4."
+        description: "Zephyr SDK version. Ex. 0.17.1."
         required: true
         type: string
-        default: 0.16.4
+        default: 0.17.1
       BOARD:
         description: "Target board."
         required: true

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -28,7 +28,7 @@ jobs:
 
     uses: ./.github/workflows/build.yml
     with:
-      ZEPHYR_SDK: 0.16.4
+      ZEPHYR_SDK: 0.17.1
       BOARD: nucleo_c031c6
 
   upload-binaries:

--- a/west.yml
+++ b/west.yml
@@ -15,5 +15,5 @@ manifest:
         path-prefix: deps
         name-allowlist:
           - zephyr
-          - cmsis
+          - cmsis_6
           - hal_stm32

